components/properties/src/error.rs

```diff
@@ -28,10 +29,23 @@ pub enum PropertiesError {
     /// An unknown value was used for the [`GeneralCategoryGroup`](crate::GeneralCategoryGroup) property
     #[displaydoc("Unknown general category group: {0}")]
     UnknownGeneralCategoryGroup(u32),
+    /// An unknown or unexpected property was used for an API dealing with properties at runtime
+    #[displaydoc("Unexpected or unknown property {0}")]
+    UnexpectedProperty(TinyStr16),
```

Comment:
Suggestion: do one of the following

1. Change this to `UnexpectedPropertyName` and adjust the docs around that
2. Make the function return `Result<Option<CodePointSetData>, PropertiesError>` and remove this error variant

The justification for 2 is that it is a getter and we failed to get anything, which isn't an error. It's only an error if we successfully matched but we don't have the data for it.

Reply:
Hmm, while I'm fine with that in this case (and will make the change); I'm not sure I agree with the justification in general. It's only a _data error_ if we successfully matched but don't have data; there are other ways to get errors.

Should this return DataError instead?